.circleci/config.yml

```
@@ -0,0 +1,247 @@
+  source_build: &source_build
+    run:
+      name: Source Build
+      command: |
+        if [ -d install ]
+        then
+            echo "Skipping Source Build"
+        else
+            . $ROS_WS/install/setup.sh
+            colcon build \
+              --symlink-install \
+              --cmake-args \
+                -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+                -DCOVERAGE_ENABLED=$COVERAGE_ENABLED
+        fi
+  save_build_cache: &save_build_cache
+    save_cache:
+      name: Save Build Cache
+      key: build-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "/opt/moveit2_ws/checksum.txt" }}
+      paths:
+        - /opt/moveit2_ws
+  copy_build_logs: &copy_build_logs
+    run:
+        name: Copy Build Logs
+        when: always
+        command: cp -rH log/latest_build log/build
+  store_build_logs: &store_build_logs
+    store_artifacts:
+        path: log/build
+  test_build: &test_build
+    run:
+        name: Test Build
+        command: |
+          . install/setup.sh
+          src/moveit2/tools/run_test_suite.bash
+  copy_test_logs: &copy_test_logs
+    run:
+        name: Copy Test Logs
+        when: always
+        command: cp -rH log/latest_test log/test
+  store_test_logs: &store_test_logs
+    store_artifacts:
+        path: log/test
+  report_code_coverage: &report_code_coverage
+    run:
+        name: Report Code Coverage
+        command: |
+          if [ "$COVERAGE_ENABLED" = "True" ]
+          then
+            . install/setup.sh
+            src/moveit2/moveit/scripts/code_coverage_report.bash codecovio
+          else
+            echo "Skipping Code Coverage Report"
+          fi
+
+commands:
+  checkout_source:
+    description: "Checkout Source"
+    steps:
+      - *pre_checkout
+      - checkout:
+          path: src/moveit2
+      - *post_checkout
+  setup_upstream:
+    description: "Setup Upstream"
+    steps:
+      - *install_upstream
+      - *restore_upstream_cache
+      - *build_upstream
+      - *save_upstream_cache
+  build_source:
+    description: "Build Source"
+    steps:
+      - *install_dependencies
+      - *store_build_checksum
+      - *restore_build_cache
+      - *source_build
+      - *save_build_cache
+      - *copy_build_logs
+      - *store_build_logs
+  test_source:
+    description: "Test Source"
+    steps:
+      - *test_build
+      - *copy_test_logs
+      - *store_test_logs
+  report_coverage:
+    description: "Report Coverage"
+    steps:
+      - *report_code_coverage
+
+executors:
+  docker_exec:
+    docker:
+      - image: moveit/moveit2:master
+    working_directory: /opt/moveit2_ws
+    environment:
+      MAKEFLAGS: "-j 1 -l 1"
+
+jobs:
+  debug_build:
+    executor: docker_exec
+    environment:
+      CMAKE_BUILD_TYPE: "Debug"
+      COVERAGE_ENABLED: "True"
+    steps:
+      - checkout_source
+      - setup_upstream
+      - build_source
+      - test_source
+      # - report_coverage
```

```suggestion
      # TODO: re-enable report coverage when...
      # - report_coverage
```
when? or remove comment

---

Is code coverage working now? I was waiting on this until this repo uses something like nav2's cmake changes to enable codecov at build time.